android/KMEA/app/src/main/java/com/keyman/engine/KeyboardPickerActivity.java

```diff
@@ -263,18 +263,6 @@ protected void onResume() {
       return;
   }
 
-  @Override
-  protected void onPause() {
-    super.onPause();
-
-    if (KMManager.InAppKeyboard != null) {
-      KMManager.InAppKeyboard.loadKeyboard();
-    }
-    if (KMManager.SystemKeyboard != null) {
-      KMManager.SystemKeyboard.loadKeyboard();
-    }
-  }
```

Comment:
Just checking that this won't cause issues with installation / removal of keyboards?

Reply:
That should be handled by the Settings menu or equivalent.  The picker doesn't touch that stuff anymore.

I'll see if I can find the part that actually _does_ handle it, but I'm having trouble remembering exactly where to look.

Reply:
... I _think_ this is it?

https://github.com/keymanapp/keyman/blob/2a9a9e7244d5753ee0a66b379220abc3af669cb4/android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java#L1742-L1751

Called by:

https://github.com/keymanapp/keyman/blob/2a9a9e7244d5753ee0a66b379220abc3af669cb4/android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java#L284-L288

I haven't yet been able to confirm whether or not update installations pass through this codeblock though; the block above is the _only_ call I can find to the method.

Reply:
> That should be handled by the Settings menu or equivalent. The picker doesn't touch that stuff anymore. 

The Keyboard picker menu still has the "hidden-menu" feature that's always been in the app. If you long-press on a keyboard, you'll get the option to "Delete" the keyboard.

Reply:
What handles resetting a keyboard after a keyboard-update operation?  I couldn't confirm that pathway.

Reply:
Good catch re: the "delete" the keyboard bit, btw.

Reply:
After a bit of guidance in other channels, I've found the spot that _should_ handle it for keyboard updates.  Not that it does, though...

https://github.com/keymanapp/keyman/blob/7b6c57d818676428a3db0f45b555420e950534a9/android/KMEA/app/src/main/java/com/keyman/engine/logic/ResourcesUpdateTool.java#L563-L582

That 'else' should trigger `KMManager.clearKeyboardCache()`, I think... though I suppose it might be annoying to have it "go off" in the middle of typing if you forget that the update was processing.

Reply:
So... yeah, the current app _is_ lazy-relying on the picker to trigger the refresh, I guess.  (Not even "any keyboard change", like a globe quick-tap - but on actual use of the picker itself.)